### PR TITLE
🎉 🐛 Source Amplitude: enable `high` strictness + fix future dates bug

### LINKED ISSUE #19036: Source Amplitude: enable `high` test strictness level in SAT
## What
A `test_strictness_level` field was introduced to Source Acceptance Tests (SAT).
Amplitude is a generally_available connector, we want it to have a `high` test strictness level.

**This will help**:
- maximize the SAT coverage on this connector.
- document its potential weaknesses in term of test coverage.

## How
1. Migrate the existing `acceptance-test-config.yml` file to the latest configuration format. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/source-acceptance-test/README.md#L61))
2. Enable `high` test strictness level in `acceptance-test-config.yml`. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/docs/connector-development/testing-connectors/source-acceptance-tests-reference.md#L240))
3. Commit changes on `acceptance-test-config.yml` and open a PR.
4. Run SAT with the `/test` command on the branch.
5. If tests are failing please fix the failing test or use `bypass_reason` fields to explain why a specific test can't be run.

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving: https://github.com/airbytehq/airbyte/issues/19036

## 🚨 User Impact 🚨
No impact is expected.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Comment by bazarnov** [comment]
Backing up myself : need to publish before merging!

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml [hunk 1]
```diff
@@ -90,6 +90,9 @@
   icon: amplitude.svg
   sourceType: api
   releaseStage: generally_available
+  allowedHosts:
+    hosts:
+      - "*.amplitude.com"
```
Comment (evantahler) [inline airbyte-config/init/src/main/resources/seed/source_definitions.yaml:95]: 👍 allowed hosts

File: airbyte-integrations/connectors/source-amplitude/acceptance-test-config.yml [hunk 2]
```diff
@@ -1,24 +1,35 @@
...
-      # future_state_path: "integration_tests/abnormal_state.json"
+    tests:
+      - config_path: "secrets/config.json"
+        expect_records:
+          path: "integration_tests/expected_records.jsonl"
+        empty_streams:
+          - name: "cohorts"
+            bypass_reason: "This stream is empty due to free subscription plan for the sandbox."
+          - name: "annotations"
+            bypass_reason: "This stream is empty due to free subscription plan for the sandbox."
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-amplitude/acceptance-test-config.yml:25]: not blocking, but what would it cost to have a plan that allows us to use these streams?

File: airbyte-integrations/connectors/source-amplitude/acceptance-test-config.yml [hunk 3]
```diff
@@ -1,24 +1,35 @@
 connector_image: airbyte/source-amplitude:dev
-tests:
+test_strictness_level: high
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-amplitude/acceptance-test-config.yml:2]: 👍 test_strictness_level

**Review by evantahler — APPROVED** [review]
expected_records looks good too!

File: airbyte-integrations/connectors/source-amplitude/acceptance-test-config.yml (hunk 2 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-amplitude/acceptance-test-config.yml:25]: @igrankova do we have the answer to this question?

Comment (igrankova) [reply airbyte-integrations/connectors/source-amplitude/acceptance-test-config.yml:25]: @evantahler, these 2 streams are on the paid plan and we are working on obtaining access to them. I'll let you know when we receive it.
cc @bazarnov, @YowanR

**Review by alafanechere — APPROVED** [review]
I'm approving the changes made to acceptance test configuration.

**Review by lazebnyi — APPROVED** [review]

**Pull request merged**